cmake/modules/RootMacros.cmake

```diff
@@ -553,8 +553,17 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
     target_compile_definitions(${dictionary} PRIVATE
       ${definitions} $<TARGET_PROPERTY:${ARG_MODULE},COMPILE_DEFINITIONS>)
 
+    # remove all -I prefixes from list of include dirs
+    # otherwise generated make files will look like:
+    # CXX_INCLUDES = -I/home/user/git/root6/io/io/-I/home/user/git/root6 ...
```

Comment:
Sorry for being slow but I still fail to understand what is wrong with `-I/home/user/git/root6/io/io/-I/home/user/git/root6`. Is it the fact that there is no `inc` suffix or the fact that there is no space between `-I` or something else?

Reply:
While cmake try to use directory `/home/user/git/root6/io/io/-I/home/user/git/root6` as includes and for dependency generation.

Reply:
As in: `target_include_directories` expects a list of directories, but before Sergey's change got a list of `-I/some/dir`s. Not good.

Reply:
The question to be asked should be why is there no space in the list that was constructed in the first place? It's usually a side-effect from mixing `list($LIST APPEND ...)` and `set(LIST $LIST $EXTRA_STUFF)`. We should first try to fix this underlying issue in my opinion.

Reply:
> The question to be asked should be why is there no space in the list

Problem that path starts with "-I" and cmake just prepend current source path. 
In no case one should call `target_include_directories()` providing such dir names. 
It is bug and I try fix it